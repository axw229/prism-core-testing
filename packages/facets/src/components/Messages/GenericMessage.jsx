// @flow
import React, { Children } from 'react'
import './GenericMessage.scss'

type Props = {
  children: any,
  type: string,
  className?: string,
  fillParent?: boolean
}

const types = {
  ERROR: 'ERROR',
  WARNING: 'WARNING',
  NORMAL: 'NORMAL',
  SUCCESS: 'SUCCESS'
}

const baseClass = 'generic-message'

const outerClasses = {
  [types.NORMAL]: `${baseClass}`,
  [types.WARNING]: `${baseClass} generic-message--warning`,
  [types.ERROR]: `${baseClass} generic-message--error`,
  [types.SUCCESS]: `${baseClass} generic-message--success`
}

const innerClasses = {
  [types.NORMAL]: `${baseClass}__inner`,
  [types.WARNING]: `${baseClass}__inner ${baseClass}__inner--warning`,
  [types.ERROR]: `${baseClass}__inner ${baseClass}__inner--error`,
  [types.SUCCESS]: `${baseClass}__inner ${baseClass}__inner--success`
}

const lineClasses = {
  BASE: `${baseClass}__line`,
  FIRST_LINE: `${baseClass}__line--headline`
}

const GenericMessage = (props: Props) => {
  const { className, children, type, fillParent, ...other } = props
  const multiChildren = Children.toArray(children).length > 1
  let mainClasses = `${outerClasses[type]} ${typeof className === 'string' ? className : ''} ${multiChildren ? `${baseClass}--multi-line` : ''}`.trim()

  if (fillParent) {
    mainClasses += '--fill-parent'
  }

  return (
    <div className={mainClasses} {...other}>
      <div className={innerClasses[type]}>
        {Children.toArray(children).map((child, i) => (
          <div key={i} className={`${lineClasses.BASE} ${i === 0 ? lineClasses.FIRST_LINE : ''}`}>{child}</div>
        ))}
      </div>
    </div>
  )
}

GenericMessage.defaultProps = {
  type: types.NORMAL
}

GenericMessage.TYPES = types

export default GenericMessage
