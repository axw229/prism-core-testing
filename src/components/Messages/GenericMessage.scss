.generic-message {
  font-size: inherit;
  padding: 1em;
  text-align: center;

  &__inner {
    font-weight: bold;
    background: rgba(pick-visible-color($color-near-black, $color-black, $color-white), $color-near-opacity);
    border-radius: .25em;
    color: $color-near-black;
    text-align: center;
    display: inline-block;
    padding: 1em;

    &--error {
      background: rgba(getVar('colors.error'), $color-near-opacity);
      color: pick-visible-color(getVar('colors.error'), $color-black, $color-white);
    }

    &--warning {
      background: rgba(getVar('colors.warning'), $color-near-opacity);
      color: pick-visible-color(getVar('colors.warning'), $color-black, $color-white);
    }
  }
}