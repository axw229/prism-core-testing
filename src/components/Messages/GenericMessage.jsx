// @flow
import React, { Children } from 'react'

import './GenericMessage.scss'

type Props = {
  children: any,
  type: string,
  className?: string
}

const types = {
  ERROR: 'ERROR',
  WARNING: 'WARNING',
  NORMAL: 'NORMAL'
}

const baseClass = 'generic-message'

const outerClasses = {
  [types.NORMAL]: `${baseClass}`,
  [types.WARNING]: `${baseClass} generic-message--warning`,
  [types.ERROR]: `${baseClass} generic-message--error`
}

const innerClasses = {
  [types.NORMAL]: `${baseClass}__inner`,
  [types.WARNING]: `${baseClass}__inner ${baseClass}__inner--warning`,
  [types.ERROR]: `${baseClass}__inner ${baseClass}__inner--error`
}

const lineClasses = {
  BASE: `${baseClass}__line`,
  FIRST_LINE: `${baseClass}__line--headline`
}

const GenericMessage = (props: Props) => {
  const { className, children, type, ...other } = props
  const multiChildren = Children.toArray(children).length > 1
  const mainClasses = `${outerClasses[type]} ${typeof className === 'string' ? className : ''} ${multiChildren ? `${baseClass}--multi-line` : ''}`

  return (
    <div className={mainClasses} {...other}>
      <div className={innerClasses[type]}>
        {Children.toArray(children).map((child, i) => (
          <div key={i} className={`${lineClasses.BASE} ${i === 0 ? lineClasses.FIRST_LINE : ''}`}>{child}</div>
        ))}
      </div>
    </div>
  )
}

GenericMessage.defaultProps = {
  type: types.NORMAL
}

GenericMessage.TYPES = types

export default GenericMessage
