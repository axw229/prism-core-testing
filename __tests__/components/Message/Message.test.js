import React from 'react'
import { shallow } from 'enzyme'
import Messages from 'src/components/Messages/GenericMessage'

const createMessage = (props = {}) => {
  return shallow(<Messages {...props} />)
}

const dummyComponent = () => {
  return <div>TEST</div>
}

// rendering and snapshot test
describe('Coordinary Color Swatch component will pass props correctly', () => {
  it('snapshot testing', () => {
    const wrapper = createMessage()
    expect(wrapper).toMatchSnapshot()
  })

  it('child component will be rendering', () => {
    const wrapper = createMessage({ children: dummyComponent })
    const childWrapper = shallow(wrapper.props().children.props.children())
    expect(childWrapper.find('div').text()).toEqual('TEST')
  })
})
