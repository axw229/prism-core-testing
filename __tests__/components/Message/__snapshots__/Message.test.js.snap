// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coordinary Color Swatch component will pass props correctly snapshot testing 1`] = `
<div
  className="generic-message "
>
  <div
    className="generic-message__inner"
  />
</div>
`;
